sympy/sets/sets.py

```diff
@@ -1063,16 +1064,18 @@ def _boundary(self):
         return FiniteSet(*finite_points)
 
     def _contains(self, other):
-        if not isinstance(other, Expr) or (
+        if (not isinstance(other, Expr) or
                 other is S.Infinity or
                 other is S.NegativeInfinity or
                 other is S.NaN or
-                other is S.ComplexInfinity) or other.is_extended_real is False:
+                other is S.ComplexInfinity or
+                other.is_extended_real is False or
+                other.is_real is False):
             return false
 
         if self.start is S.NegativeInfinity and self.end is S.Infinity:
-            if not other.is_extended_real is None:
-                return other.is_extended_real
+            if not other.is_real is None:
+                return other.is_real
```

Comment:
`x is not None` is better than `not x is None` as there is no ambiguity in the precedence